src/wazuh_modules/syscollector/include/syscollector.hpp

```diff
@@ -100,11 +100,13 @@ class EXPORTED Syscollector final
         void scan();
         void sync();
         void syncLoop(std::unique_lock<std::mutex>& lock);
+        void syncAlgorithm();
         std::shared_ptr<ISysInfo>                                               m_spInfo;
         std::function<void(const std::string&)>                                 m_reportDiffFunction;
         std::function<void(const std::string&)>                                 m_reportSyncFunction;
         std::function<void(const modules_log_level_t, const std::string&)>      m_logFunction;
-        unsigned int                                                            m_intervalValue;
+        unsigned int                                                       m_intervalValue;
+        unsigned int                                                       m_currentIntervalValue;
```

Comment:
Use std::chrono::seconds
```suggestion
        std::chrono::seconds                                                       m_intervalValue;
        std::chrono::seconds                                                       m_currentIntervalValue;
```